idl/forward_request.idl.hh

```diff
@@ -0,0 +1,23 @@
+namespace query {
+struct forward_request {
+    enum class reduction_type : uint8_t {
+        count,
+    };
+
+    std::vector<query::forward_request::reduction_type> reduction_types;
+    std::vector<column_id> selected_columns;
+
+    query::read_command cmd;
+    dht::partition_range_vector pr;
+
+    db::consistency_level cl;
+    lowres_clock::time_point timeout;
+    bool skip_metadata;
```

Comment:
Why do we need skip_metadata?

Reply:
@havaker the super-coordinator is supposed to provide the metadata, right? I can see something like this near the end of the `forwarding_select_statement::do_execute` function. If so, then this field shouldn't be needed (I think you always set it to `true` at the moment).

Reply:
When writing the first version of this pr, I was trying to send (from a super-coordinator to a coordinator) all objects required to create `pager`. Among those objects, there was `cql3::query_options`. I couldn't serialize it easily, so I chose to send some of its members. One of these members was `skip_metadata`.

@piodul you are right. It's the super-coordinator's job to provide the metadata. [It does so here](https://github.com/scylladb/scylla/pull/9209/commits/12c32c87dc9a40ea2ffc3c7af315c64dfe5fcc9a#diff-1073ea9ce4c5e00bb6eb614154f523ba7962403a4fe6c8cd877d1c8b73b3f649R1445).
Also, I found out, that `cql3::query_options::skip_metadata()` isn't used inside `do_execute` methods of select statements. It's used a layer of abstraction lower, (`transport/server.cc`) while serializing responses. And so, sending it with `forward_request` to the coordinators is redundant, because it does not affect `forward_result`s produced by coordinators.

I will remove this field it in the next version.